contracts/adapters/FlexibleLeverageStrategyAdapter.sol

```diff
@@ -638,14 +642,28 @@ contract FlexibleLeverageStrategyAdapter is BaseAdapter {
             newLeverageRatio = _calculateNewLeverageRatio(currentLeverageRatio);
         }
 
-        ActionInfo memory actionInfo = _createActionInfo();
+        sizes = new uint256[](_exchangeNames.length);
+        isLevers = new bool[](_exchangeNames.length);
+        for (uint256 i = 0; i < _exchangeNames.length; i++) {
+
+            ExchangeSettings memory settings = exchangeSettings[_exchangeNames[i]];
+            uint256 maxTradeSize = isRipcord ? settings.incentivizedTwapMaxTradeSize : settings.twapMaxTradeSize;
+            uint256 slippageTolerance = isRipcord ? incentive.incentivizedSlippageTolerance : execution.slippageTolerance;
+        
+            bool isLever = newLeverageRatio > currentLeverageRatio;
+            LeverageInfo memory leverageInfo = _getAndValidateLeveragedInfo(slippageTolerance, maxTradeSize, _exchangeNames[i]);
 
-        bool isLever = newLeverageRatio > currentLeverageRatio;
-        uint256 leverageRatioDifference = isLever ? newLeverageRatio.sub(currentLeverageRatio) : currentLeverageRatio.sub(newLeverageRatio);
+            (uint256 collateralNotional,) = _calculateChunkRebalanceNotional(leverageInfo, newLeverageRatio, isLever);
 
-        uint256 collateralRebalanceNotional = leverageRatioDifference.preciseDiv(currentLeverageRatio).preciseMul(actionInfo.collateralBalance);
+            if (isLever) {
+                ActionInfo memory actionInfo = _createActionInfo();
+                sizes[i] = _calculateBorrowUnits(collateralNotional, actionInfo);
```

Comment:
This should be `_calculateMaxBorrowCollateral`? I think we want to always be passing along the same asset

Reply:
Actuall I don't think we need this logic. What's the thought behind adding this if statement? `_calculateChunkRebalanceNotional` already calculates the max borrow amount